src/librustc_mir/interpret/eval_context.rs

```diff
@@ -1211,115 +1235,56 @@ impl<'a, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> EvalContext<'a, 'mir, 'tcx, M
-                self.memory.write_primval(a_ptr, dest_align, a_val, a_size, false)?;
-                self.memory.write_primval(b_ptr, dest_align, b_val, b_size, false)
+                self.memory.write_scalar(a_ptr, dest_align, a_val, a_size, false)?;
+                self.memory.write_scalar(b_ptr, dest_align, b_val, b_size, false)
             }
         }
     }
 
-    pub fn ty_to_primval_kind(&self, ty: Ty<'tcx>) -> EvalResult<'tcx, PrimValKind> {
-        use syntax::ast::FloatTy;
-
-        let kind = match ty.sty {
-            ty::TyBool => PrimValKind::Bool,
-            ty::TyChar => PrimValKind::Char,
-
-            ty::TyInt(int_ty) => {
-                use syntax::ast::IntTy::*;
-                let size = match int_ty {
-                    I8 => Size::from_bytes(1),
-                    I16 => Size::from_bytes(2),
-                    I32 => Size::from_bytes(4),
-                    I64 => Size::from_bytes(8),
-                    I128 => Size::from_bytes(16),
-                    Isize => self.memory.pointer_size(),
-                };
-                PrimValKind::from_int_size(size)
-            }
-
-            ty::TyUint(uint_ty) => {
-                use syntax::ast::UintTy::*;
-                let size = match uint_ty {
-                    U8 => Size::from_bytes(1),
-                    U16 => Size::from_bytes(2),
-                    U32 => Size::from_bytes(4),
-                    U64 => Size::from_bytes(8),
-                    U128 => Size::from_bytes(16),
-                    Usize => self.memory.pointer_size(),
-                };
-                PrimValKind::from_uint_size(size)
-            }
-
-            ty::TyFloat(FloatTy::F32) => PrimValKind::F32,
-            ty::TyFloat(FloatTy::F64) => PrimValKind::F64,
-
-            ty::TyFnPtr(_) => PrimValKind::FnPtr,
-
-            ty::TyRef(_, ty, _) |
-            ty::TyRawPtr(ty::TypeAndMut { ty, .. }) if self.type_is_sized(ty) => {
-                PrimValKind::Ptr
-            }
-
-            ty::TyAdt(def, _) if def.is_box() => PrimValKind::Ptr,
-
-            ty::TyAdt(..) => {
-                match self.layout_of(ty)?.abi {
-                    layout::Abi::Scalar(ref scalar) => {
-                        use rustc::ty::layout::Primitive::*;
-                        match scalar.value {
-                            Int(i, false) => PrimValKind::from_uint_size(i.size()),
-                            Int(i, true) => PrimValKind::from_int_size(i.size()),
-                            F32 => PrimValKind::F32,
-                            F64 => PrimValKind::F64,
-                            Pointer => PrimValKind::Ptr,
-                        }
-                    }
-
-                    _ => return err!(TypeNotPrimitive(ty)),
-                }
-            }
-
-            _ => return err!(TypeNotPrimitive(ty)),
-        };
-
-        Ok(kind)
+    pub fn ty_to_primitive(&self, ty: Ty<'tcx>) -> EvalResult<'tcx, layout::Primitive> {
```

Comment:
This is only used in one function, can you inline it there? Seems very niche and potentially misusable.